scala/private/rules/scala_test.bzl

```diff
@@ -0,0 +1,225 @@
+        unused_dependency_checker_is_off = unused_dependency_checker_is_off,
+    )
+    (
+        cjars,
+        transitive_rjars,
+        transitive_compile_jars,
+        jars_to_labels,
+    ) = (
+        jars.compile_jars,
+        jars.transitive_runtime_jars,
+        jars.transitive_compile_jars,
+        jars.jars2labels,
+    )
+
+    args = "\n".join([
+        "-R",
+        ctx.outputs.jar.short_path,
+        _scala_test_flags(ctx),
+        "-C",
+        "io.bazel.rules.scala.JUnitXmlReporter",
+    ])
+
+    argsFile = ctx.actions.declare_file("%s.args" % ctx.label.name)
+    ctx.actions.write(argsFile, args)
+
+    executable = declare_executable(ctx)
+
+    wrapper = write_java_wrapper(ctx, "", "")
+    out = scala_binary_common(
+        ctx,
+        executable,
+        cjars,
+        transitive_rjars,
+        transitive_compile_jars,
+        jars_to_labels,
+        wrapper,
+        unused_dependency_checker_ignored_targets =
+            unused_dependency_checker_ignored_targets,
+        unused_dependency_checker_mode = unused_dependency_checker_mode,
+        runfiles_ext = [argsFile],
+        deps_providers = jars.deps_providers,
+    )
+
+    rjars = out.transitive_rjars
+
+    coverage_runfiles = []
+    if ctx.configuration.coverage_enabled and _coverage_replacements_provider.is_enabled(ctx):
+        coverage_replacements = _coverage_replacements_provider.from_ctx(
+            ctx,
+            base = out.coverage.replacements,
+        ).replacements
+
+        rjars = depset([
+            coverage_replacements[jar] if jar in coverage_replacements else jar
+            for jar in rjars.to_list()
+        ])
+        coverage_runfiles = ctx.files._jacocorunner + ctx.files._lcov_merger + coverage_replacements.values()
+
+    # jvm_flags passed in on the target override scala_test_jvm_flags passed in on the
+    # toolchain
+    final_jvm_flags = first_non_empty(
+        ctx.attr.jvm_flags,
+        ctx.toolchains["@io_bazel_rules_scala//scala:toolchain_type"].scala_test_jvm_flags,
+    )
+
+    coverage_runfiles.extend(write_executable(
+        ctx = ctx,
+        executable = executable,
+        jvm_flags = [
+            "-DRULES_SCALA_MAIN_WS_NAME=%s" % ctx.workspace_name,
+            "-DRULES_SCALA_ARGS_FILE=%s" % argsFile.short_path,
+        ] + expand_location(ctx, final_jvm_flags),
+        main_class = ctx.attr.main_class,
+        rjars = rjars,
+        use_jacoco = ctx.configuration.coverage_enabled,
+        wrapper = wrapper,
+    ))
+
+    return struct(
+        executable = executable,
+        files = out.files,
+        instrumented_files = out.instrumented_files,
+        providers = out.providers,
+        runfiles = ctx.runfiles(coverage_runfiles, transitive_files = out.runfiles.files),
+        scala = out.scala,
+    )
+
+_scala_test_attrs = {
+    "main_class": attr.string(
+        default = "io.bazel.rulesscala.scala_test.Runner",
+    ),
+    "suites": attr.string_list(),
```

Comment:
can we remove this also? We had it deprecated for years now with a print nag warning. I think it should be safe to remove.